go/contacts/resolution_notice.go

```diff
@@ -0,0 +1,145 @@
+package contacts
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"github.com/keybase/keybase/go/base/codec"
+
+	"github.com/keybase/client/go/encrypteddb"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type ContactResolution struct {
+	Description  string
+	ResolvedUser keybase1.User
+}
+
+const ContactResolutionEncryptionVersion = 1
+
+type ContactResolutionForEncryptrion struct {
+	ContactResolution
+	EncryptionVersion int
+}
+
+type EncryptedContactResolution struct {
+	Blob          []byte
+	PukGeneration keybase1.PerUserKeyGeneration
+}
+
+var errWrongContactEncryptionScheme = errors.New(
+	"wrong encryption version for encrypted contact blob")
+
+func getKeyFn(mctx libkb.MetaContext,
```

Comment:
```suggestion
func getPerUserKeyFn(mctx libkb.MetaContext,
```

Reply:
can we add a test that does this for two users and ensures secret is different? to prevent bugs where the key is all zeros for some reason